docs/3.0-migration-guide.md

```diff
@@ -65,8 +509,38 @@ If you're a library user and used to rely on having access to a structure or
 function that's now in a private header, please reach out on the mailing list
 and explain your need; we'll consider adding a new API in a future version.
 
-Remove the certs module from the library
-----------------------------------------
+### CCM interface changes: impact for alternative implementations
+
+The CCM interface has changed with the addition of support for
+multi-part operations. Five new API functions have been defined:
+mbedtls_ccm_starts(), mbedtls_ccm_set_lengths(),
+mbedtls_ccm_update_ad(), mbedtls_ccm_update() and mbedtls_ccm_finish().
+Alternative implementations of CCM (`MBEDTLS_CCM_ALT`) have now to
+implement those additional five API functions.
+
+
+
+## X.509
+
+### Strengthen default algorithm selection for X.509 and TLS
```

Comment:
I think most users will encounter this via TLS — even if they're affected by the X.509 changes, it's because of TLS connections. So I think this section should go under the TLS chapter. But it would make sense to put a note “see also …” at the top of the X.509 chapter. (And likewise for some mixed low/high-level crypto sections, by the way.)

Reply:
I've made a single generic cross-reference for the low/high-level items, as otherwise it might be hard to draw a line as to which we should cross-reference.